### PR TITLE
RCPS-64 Fix View Violation in Prefect System

### PR DESCRIPTION
**Pull request by daiahnadine**

### CODE REVIEW

File: logs/prefect-office-record-management-system-20240327.log
Comment (RoCS2024) [inline logs/prefect-office-record-management-system-20240327.log:1]: remove logs folder in the commit

File: src/main/java/Main.java [hunk 1]
```diff
@@ -190,6 +196,26 @@ private static void addViolation() {
...
+                    System.out.println("Violation ID: " + violationRecord.getViolation());
+                    System.out.println("Violation Type: " + violationRecord.getType());
+                    System.out.println("Community Service Hours: " + violationRecord.getCommServHours());
+                    System.out.println("-----------------------------------");
+                }
+            } else {
+                System.out.println("No violation records found.");
+            }
+        } catch (Exception e) {
+            System.err.println("An error occurred while viewing all violation: " + e.getMessage());
```
Comment (RoCS2024) [inline src/main/java/Main.java:216]: why use System.err?

File: src/main/java/com/prefect/office/record/management/app/facade/prefect/violation/impl/ViolationFacadeImpl.java [hunk 2]
```diff
@@ -2,11 +2,24 @@
 
 import com.prefect.office.record.management.app.facade.prefect.violation.ViolationFacade;
 import com.prefect.office.record.management.app.model.violation.Violation;
+import com.prefect.office.record.management.data.connectionhelper.ConnectionHelper;
 import com.prefect.office.record.management.data.dao.prefect.violation.ViolationDao;
 import com.prefect.office.record.management.data.dao.prefect.violation.impl.ViolationDaoImpl;
+import oracle.jdbc.proxy.annotation.Pre;
```
Comment (RoCS2024) [inline src/main/java/com/prefect/office/record/management/app/facade/prefect/violation/impl/ViolationFacadeImpl.java:8]: what is this for?

File: src/main/java/com/prefect/office/record/management/app/facade/prefect/violation/impl/ViolationFacadeImpl.java [hunk 3]
```diff
@@ -2,11 +2,24 @@
...
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.util.ArrayList;
+import java.util.List;
 
 
 public class ViolationFacadeImpl implements ViolationFacade {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(ViolationFacade.class);
+    Connection c = ConnectionHelper.getConnection();
```
Comment (RoCS2024) [inline src/main/java/com/prefect/office/record/management/app/facade/prefect/violation/impl/ViolationFacadeImpl.java:22]: Do not make global connection. this has to be closed everytime it gets open. either:
1. localize the use of connection the close it afterwards
or 
2. use try with resources

File: src/main/main.iml
Comment (RoCS2024) [inline src/main/main.iml:1]: remove this file from the commit

File: src/test/test.iml
Comment (RoCS2024) [inline src/test/test.iml:1]: remove this file from the commit

**Review by RoCS2024 — CHANGES_REQUESTED** [review]
see comments

**Comment by daiahnadine** [comment]
These are noted po, Sir.

File: src/main/java/com/prefect/office/record/management/app/facade/prefect/violation/impl/ViolationFacadeImpl.java (hunk 2 above)
Comment (RoCS2024) [reply src/main/java/com/prefect/office/record/management/app/facade/prefect/violation/impl/ViolationFacadeImpl.java:8]: @daiahnadine did you address this?

**Pull request closed without merging**